### ISSUE TITLE
tdnf can't cope with RPM repos that do not contain a `filelists`

### ISSUE THREAD

**Issue by andrew-lee-1089** [vmware/tdnf#273]
I have two RPM repos (created by artifactory), comapring the repodata/repomd.xml , in the working case we have:

```
<?xml version="1.0"?>
<repomd xmlns="http://linux.duke.edu/metadata/repo" xmlns:rpm="http://linux.duke.edu/metadata/rpm">
  <data type="other">
    ...
  </data>
  <data type="filelists">
    ...
  </data>
  <data type="primary">
    ...
  </data>
</repomd>
```

Failing case:

```
<?xml version="1.0"?>
<repomd xmlns="http://linux.duke.edu/metadata/repo" xmlns:rpm="http://linux.duke.edu/metadata/rpm">
  ​<data type="other">
    ​...
  </data>
  <data type="primary">
    ...
  </data>
</repomd>
``` 

Looking at the code in https://github.com/vmware/tdnf/blob/v2.1.0/client/repo.c#L878:

```
    dwError = TDNFFindRepoMDPart(
                  pRepo,
                  TDNF_REPOMD_TYPE_FILELISTS,
                  &pRepoMD->pszFileLists);
    BAIL_ON_TDNF_ERROR(dwError);

    dwError = TDNFFindRepoMDPart(
                  pRepo,
                  TDNF_REPOMD_TYPE_UPDATEINFO,
                  &pRepoMD->pszUpdateInfo);
    //updateinfo is not mandatory
    if(dwError == ERROR_TDNF_NO_DATA)
    {
        dwError = 0;
    }
    BAIL_ON_TDNF_ERROR(dwError);
```

It looks like tdnf can't cope with this. Both yum and dnf can cope with this. When I last looked at it ,the specs for what exactly should be in a repomd.xml file were not clear (i.e. http://linux.duke.edu/metadata/rpm is a dead link) - but I think I would welcome a defensive fix in tdnf to cope with this. We already deem 'updateinfo' to not be mandatory.

**Comment by oliverkurth** [comment]
Thank you for reporting this.

Do you happen to know a way to reproduce a repo without a filelist, other than editing the `repomd.xml` file?

The fix is easy, but I would like to write a test for it.

**Comment by andrew-lee-1089** [comment]
The way I did was by creating a brand new RPM repo using JFROG's artifactory. I don't know if you have access to that.

**Comment by keerthanakalyan** [comment]
> Thank you for reporting this.
> 
> Do you happen to know a way to reproduce a repo without a filelist, other than editing the `repomd.xml` file?
> 
> The fix is easy, but I would like to write a test for it.

Modifyrepo_c command is used to change any metadata from the existing repo.
You can remove filelists and filelists_db from repodata using below command.
Example:
```
modifyrepo_c --remove filelists <path-to-repodata-folder>
modifyrepo_c --remove filelists_db <path-to-repodata-folder>
```

**Comment by oliverkurth** [comment]
Fixed with PR #275
